sdk/core/az_core/src/az_credential_client_secret.c

```diff
@@ -67,8 +67,13 @@ static AZ_NODISCARD az_result _az_credential_client_secret_apply(
 
   int16_t const token_length = token._internal.token_length;
 
+  // Still bad in terms of concurrency, but at least the pointer is not pointing to an unknown
+  // memory. Temporary measure to improve https://github.com/Azure/azure-sdk-for-c/issues/799 a bit.
+  uint8_t* const token_buf_ptr
+      = (uint8_t*)(uintptr_t)credential->_internal.token_credential._internal.token._internal.token;
+
   AZ_RETURN_IF_FAILED(az_http_request_append_header(
-      ref_request, _az_auth_header_name, az_span_init(token._internal.token, token_length)));
+      ref_request, _az_auth_header_name, az_span_init(token_buf_ptr, token_length)));
```

Comment:
Can we add unit tests to capture the bug that is being fixed?

Reply:
We'll enable live testing, it should catch that - this is how Victor caught it. Unit testing (at least in the compiler config it's built) did not catch this, because stack contents were still the same I assume.